src/main/python/deep_qa/layers/noisy_or.py

```diff
@@ -0,0 +1,88 @@
+from keras import backend as K
+from keras.layers import Layer
+from keras.initializations import identity
+from keras.constraints import Constraint, NonNeg
+from keras.regularizers import L1L2Regularizer
+from overrides import overrides
+
+class BetweenZeroAndOne(Constraint):
+    """Constrains the weights to be between zero and one
+    """
+    def __call__(self, p):
+        # handle negatives
```

Comment:
By "handle negatives" here, you mean that you're clipping anything below zero to be epsilon?  I'm pretty sure that's what this logic is doing.  Complete sentences explaining what the code is doing when it's not clear are better than sentence fragments: https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar.